source/connection_manager.c

```diff
@@ -767,11 +748,9 @@ static void s_schedule_connection_culling(struct aws_http_connection_manager *ma
 
     if (manager->cull_task == NULL) {
         manager->cull_task = aws_mem_calloc(manager->allocator, 1, sizeof(struct aws_task));
-        if (manager->cull_task == NULL) {
-            return;
-        }
-
         aws_task_init(manager->cull_task, s_cull_task, manager, "cull_idle_connections");
+        /* For the task to properly run and cancel, we need to keep manager alive */
+        aws_ref_count_acquire(&manager->internal_ref_count);
```

Comment:
there's error handling at end of function that would clean up cull_task

either:
- decrement the refcount there
- OR the error is actually impossible now that allocations can't fail, so just remove error-handling altogether